### ISSUE TITLE
[NES II] Move slot select to $exxx address range

### ISSUE THREAD

**Issue by JMarlin** [JMarlin/nesii#37]
Requires prior completion of #36 so that new slot location doesn't conflict with switches
Intent is to slide up the slot area to provide more room for cart RAM

**Comment by JMarlin** [comment]
Making the steps explicit here:

- [ ] Update PLD to remap the slot location
- [ ] Update BIOS to remap the slot location
- [ ] Update NOS to remap the slot location
- [ ] Update MARIO to remap the slot location
- [ ] Update DONKEY to remap the slot location

**Comment by JMarlin** [comment]
Updated everything and started testing -- looks like I probably have a bug somewhere because I'm back to getting that disk corruption again. Time to start debugging.

**Comment by JMarlin** [comment]
As a note to self: Is there really a problem with stashing this issue for later and just moving forward with a memory map of 16k RAM, 4k slot, 8k RAM, 4k bios/switches? Jam BASIC and the floppy write routines in the high 8k and call it a day.

**Comment by JMarlin** [comment]
Um. Hmm. Did a double-check on this after lunch and it's totally possible that I just forgot to copy the new NOS image to the ADT directory because I did that and wrote a new floppy and it looks like it works well.

**Comment by JMarlin** [comment]
Whelp, NVM. Tried to launch DONKEY, but apparently the switches aren't working because after loading it booted right back into the BIOS. 

Gonna need to give this a thorough clean-out and retry in the AM.

**Comment by JMarlin** [comment]
Did some harnessing, and the switches aren't getting toggled. So I suppose that's good.
I think now I'll move on to setting breakpoints to debug NOS boot to try and figure out where it's going wrong.

**Comment by JMarlin** [comment]
Ah. Right. The issue I saw last time is that the switches AREN'T working. Right.

**Comment by JMarlin** [comment]
Weird day. Tried using an actual EEPROM in case it was that my emulator doesn't interact well with the switches. Tried increasing the monitor wait routine time just in case we were missing steps occasionally or something. Results were absolutely all over the place, so I think I need to take a bit step back. 

I did pull the motherboard out of the NES with the aim of fully instrumenting it, which I think it's about time to do. Next session, I will instrument the whole thing, roll back to current master on NOS and the BIOS, and start really debugging step-by-step.

Note that I will definitely want to instrument the full data bus, the full address bus, the RWB line, the switch clock line, the switches, and the slot device select line. I think I can do all of those, and with those fully recorded I think I should be able to get clear insight into the operation of this thing.

**Comment by JMarlin** [comment]
Creating bug #44 as a blocker on this item

**Comment by JMarlin** [comment]
#44 has been closed by building a new card out of parts that haven't been reverse biased for tens of hours. Let's carry on.

**Comment by JMarlin** [comment]
Unblocking, re-fixed #40 and confirmed MARIO is running again. Happy day. Now I HAVE to start back on this one.

**Comment by JMarlin** [comment]
Closed with #46
